### PR TITLE
[Console] improve test coverage

### PR DESCRIPTION
**Pull request by TomasVotruba**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

Mirror to #16624, just to 2.7 branch

### CODE REVIEW

File: src/Symfony/Component/Console/Command/Command.php [hunk 1]
```diff
@@ -516,7 +516,7 @@ public function getProcessedHelp()
     /**
      * Sets the aliases for the command.
      *
-     * @param string[] $aliases An array of aliases for the command
+     * @param string[]|\Traversable $aliases An array of aliases for the command
```
Comment (xabbuh) [inline src/Symfony/Component/Console/Command/Command.php]: This looks wrong to me.

Comment (TomasVotruba) [inline src/Symfony/Component/Console/Command/Command.php]: @xabbuh Based on discussion on previous PR: https://github.com/symfony/symfony/pull/16624#discussion_r45559942
If still looks wrong, could you elaborate?

Comment (xabbuh) [inline src/Symfony/Component/Console/Command/Command.php]: In PHPdoc notation `[]` matches also traversables, doesn't it?

Comment (TomasVotruba) [inline src/Symfony/Component/Console/Command/Command.php]: I'm not familiar with this. I also prefer explicit types to. `string[]` not supporting `Traversable` could be ambiguous. 

Also difference when I get `array` and when I get `Traversable`.

File: src/Symfony/Component/Console/Tests/Command/CommandTest.php.orig [hunk 2]
```diff
@@ -0,0 +1,420 @@
+<?php
```
Comment (mheki) [inline src/Symfony/Component/Console/Tests/Command/CommandTest.php.orig]: What is the purpose of this file?

File: src/Symfony/Component/Console/Tests/Command/CommandTest.php [hunk 3]
```diff
@@ -341,10 +353,68 @@ public function testLegacyAsText()
...
+        $outputMock = $this->prophesize('Symfony\Component\Console\Output\OutputInterface');
+        $command->run($inputMock->reveal(), $outputMock->reveal());
+
+        $this->assertSame('process title', \PHPUnit_Framework_Assert::getObjectAttribute($command, 'processTitle'));
+    }
+
+    /**
+     * @dataProvider getSetAliasesData()
+     *
+     * @param string[]|\Traversable $aliases
```
Comment (WouterJ) [inline src/Symfony/Component/Console/Tests/Command/CommandTest.php]: tests shouldn't have PHPdocs (only the dataProvider and expectedException annotations should be used)

File: src/Symfony/Component/Console/Tests/Command/CommandTest.php [hunk 4]
```diff
@@ -341,10 +353,68 @@ public function testLegacyAsText()
...
+        $command = new TestCommand();
+        $command->setAliases($aliases);
+
+        $this->assertEquals($aliases, $command->getAliases());
+    }
+
+    /**
+     * @return array[]
+     */
+    public function getSetAliasesData()
```
Comment (mheki) [inline src/Symfony/Component/Console/Tests/Command/CommandTest.php]: `getSet...` name makes me sad

File: src/Symfony/Component/Console/Tests/Command/CommandTest.php.orig (hunk 2 above)
Comment (TomasVotruba) [inline src/Symfony/Component/Console/Tests/Command/CommandTest.php.orig]: Not sure how it get there, probably some git history. Thanks for noticing.

File: src/Symfony/Component/Console/Tests/Command/CommandTest.php (hunk 4 above)
Comment (TomasVotruba) [inline src/Symfony/Component/Console/Tests/Command/CommandTest.php]: What would make you happy?

Comment (mheki) [inline src/Symfony/Component/Console/Tests/Command/CommandTest.php]: world peace would make me happy :smile: 
But regarding PR:
How about `getAliasesData` or `getDataForSettingAliases` or whatever, but not `getSet... go!` ;)

Comment (TomasVotruba) [inline src/Symfony/Component/Console/Tests/Command/CommandTest.php]: I picked `provideDataForSettingAliases`, as `provide*` is something common name here.

File: src/Symfony/Component/Console/Command/Command.php (hunk 1 above)
Comment (TomasVotruba) [inline src/Symfony/Component/Console/Command/Command.php]: If this should delay this PR, what do you suggest to make it right?

Using `sring[]`?

Comment (mheki) [inline src/Symfony/Component/Console/Command/Command.php]: yeah, I'd leave it as it was

**Comment by TomasVotruba** [comment]
It looks like autoloading in 5.6 does not load test fixture: https://travis-ci.org/symfony/symfony/jobs/92907283#L2845

Any idea why?

**Comment by stof** [comment]
Well, the class name is not TestCommand anymore. So it looks to me that you forgot one place using the TestCommand class name

**Comment by TomasVotruba** [comment]
@stof Thanks. I've checked for that and found nothing. That would probably break it on other builds as well. 

It leads to line: https://github.com/TomasVotruba/symfony/blob/33cd2717b4882d56c75c2e3a448307730575d322/src/Symfony/Component/Console/Tests/Command/CommandTest.php#L333, where is just an empty line.

**Comment by stof** [comment]
tests are still broken (don't consider the 5.4 and 5.5 jobs on Travis: they don't actually run tests for PRs, only for pushes)

**Comment by TomasVotruba** [comment]
I still don't know why, as I wrote in previous message,

The travis https://travis-ci.org/symfony/symfony/jobs/92916535#L2150
link to empty line https://github.com/TomasVotruba/symfony/blob/console-coverage-2.7/src/Symfony/Component/Console/Tests/Command/CommandTest.php#L333

TestCommand is used with namespace everywhere in https://github.com/TomasVotruba/symfony/blob/console-coverage-2.7/src/Symfony/Component/Console/Tests/Command/CommandTest.php

Dunno how to solve this, since test passes locally well.

File: src/Symfony/Component/Console/Command/Command.php (hunk 1 above)
Comment (fabpot) [inline src/Symfony/Component/Console/Command/Command.php]: should be reverted, we do not support Traversable here (if that happens to work, fine, but that's not supported).

File: src/Symfony/Component/Console/Command/Command.php [hunk 5]
```diff
@@ -540,7 +540,7 @@ public function setAliases($aliases)
     /**
      * Returns the aliases for the command.
      *
-     * @return array An array of aliases for the command
+     * @return string[]|\Traversable An array of aliases for the command
```
Comment (fabpot) [inline src/Symfony/Component/Console/Command/Command.php]: should be reverted as well

**Comment by fabpot** [comment]
If tests do not pass, let's close it as we won't be able to merge it anyway.

**Comment by fabpot** [comment]
@TomasVotruba I'm going to close this one. Feel free to reopen a new one, perhaps with just a subset that passes.

**Pull request closed without merging**